File: Modern/src/main/scala/prv/zielony/transaction/analyst/modern/dao/UsersDao.scala
package prv.zielony.transaction.analyst.modern.dao

import java.util.Date

import com.datastax.driver.core.Row
import com.websudos.phantom.column.DateColumn
import com.websudos.phantom.keys.PartitionKey
import prv.zielony.transaction.analyst.modern.model.User

import com.websudos.phantom.dsl._

import scala.concurrent.Future

/**
 * Created by zielony on 06.03.16.
 */
class UsersDao extends CassandraTable[AbstractUsersDao, User]{

  object id extends LongColumn(this) with PartitionKey[Long]

  object birthDate extends OptionalDateColumn(this)

  override def fromRow(row: Row): User = {
   User(id(row), birthDate(row))
  }
}

abstract class AbstractUsersDao extends UsersDao with RootConnector {
  def findAll():Future[List[Option[Date]]] = {
    select(_.birthDate).fetch
  }

  def persistUser(user:User):Future[ResultSet] = {

    insert()
      .value(_.id, user.id)
      .value(_.birthDate, user.birthDate)
      .future
  }
}

File: Classical/src/main/java/prv/zielony/transaction/analyst/classical/dao/UsersDao.java
package prv.zielony.transaction.analyst.classical.dao;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;
import prv.zielony.transaction.analyst.classical.model.User;

import java.util.List;

/**
 * Created by zielony on 27.02.16.
 */
@Repository
public interface UsersDao extends JpaRepository<User, Long> {

    @Query("SELECT u from User u WHERE u.personalData IS NOT NULL AND u.personalData.country = :country")
    List<User> getClientsByCountry(@Param("country") String country);
}
